src/Domain/Model/Skill.php

```diff
@@ -48,4 +48,23 @@ public function removeResource($resourceId)
             }
         }
     }
+
+    public function addResourceVote($resourceId, ResourceVote $resourceVote) {
```

Comment:
I would split this whole into separate function like `$this->getResource($resourceId)` and `$resource->getVoteByUser($user)`. 
Decompose this into separate smaller functions - should make it more readable